pkg/operator/webhook_configuration.go

```diff
@@ -170,10 +170,13 @@ func applyUnstructured(resourceClient dynamic.ResourceInterface, path string, re
 	}
 
 	existing, err := resourceClient.Get(context.TODO(), required.GetName(), metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		actual, err := resourceClient.Create(context.TODO(), required, metav1.CreateOptions{})
-		reportCreateEvent(recorder, required, err)
-		return actual, true, err
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			actual, err := resourceClient.Create(context.TODO(), required, metav1.CreateOptions{})
+			reportCreateEvent(recorder, required, err)
+			return actual, true, err
```

Comment:
We should check the error here before we report the event as we have done on the Update later in this file